script/make_release.py

```diff
@@ -0,0 +1,259 @@
+                    help='The maximum number of jobs to run concurrently.')
+parser.add_argument('--repository', default='https://github.com/official-stockfish/Stockfish.git', type=str,
+                    help='The path to the repository to clone.')
+parser.add_argument('--clone-dir', default='repo', type=str,
+                    help='The directory to clone the repository to.')
+parser.add_argument('--out-dir', default='bin', type=str,
+                    help='The directory that shall contain the resulting binaries.')
+parser.add_argument('--build-type', default='profile-build', type=str,
+                    help='The build type to use in make.')
+parser.add_argument('--prefix', default='', type=str,
+                    help='The string put in the binary name after stockfish- but before -arch. For example "win" or "linux". Will be ommited if empty.')
+parser.add_argument('--verbose', default=False, action='store_true',
+                    help='Keep output of running programs.')
+parser.add_argument('--keep-logs', default=False, action='store_true',
+                    help='Whether to keep logs in the output directory. Incompatible with --verbose.')
+parser.add_argument('archs', nargs='*', default=DEFAULT_ARCHS, type=str,
+                    help='A list of architectures to compile. If not provided then most sensible archs will be built.')
+
+args = parser.parse_args()
+
+if args.verbose and args.keep_logs:
+    parser.error('--verbose and --keep-logs are currently not compatible.')
+
+SDE_PATH = args.sde
+NUM_THREADS = args.concurrency
+REPO_DIR = args.clone_dir
+OUT_DIR = args.out_dir
+REPO = args.repository
+BUILD_TYPE = args.build_type
+ARCHS = args.archs
+STDOUT = None if args.verbose else subprocess.DEVNULL
+STDERR = subprocess.STDOUT
+PREFIX = args.prefix
+IS_WINDOWS = 'windows' in platform.system().lower()
+EXE_SUFFIX = '.exe' if IS_WINDOWS else ''
+
+def rmtree(top):
+    '''
+    Recursively removes the directory `top`
+    '''
+    for root, dirs, files in os.walk(top, topdown=False):
+        for name in files:
+            filename = os.path.join(root, name)
+            os.chmod(filename, stat.S_IWUSR)
+            os.remove(filename)
+        for name in dirs:
+            os.rmdir(os.path.join(root, name))
+    os.rmdir(top)
+
+def remove(path):
+    '''
+    If `path` does not point to an existing filesystem object
+    the function does nothing.
+    If `path` is a path to an existing file the file is deleted.
+    If `path` is a path to an existing directory the directory
+    and all its content is deleted
+    Otherwise a ValueError is thrown.
+    '''
+    if not os.path.exists(path):
+        return
+
+    if os.path.isfile(path) or os.path.islink(path):
+        os.remove(path)
+    elif os.path.isdir(path):
+        rmtree(path)
+    else:
+        raise ValueError("file {} is not a file or dir.".format(path))
+
+def copytree(src, dst, symlinks=False, ignore=None):
```

Comment:
IMO this function should be replaced by `shutil.copytree(src, dst)`
https://docs.python.org/3/library/shutil.html#shutil.copytree

Reply:
doesn't work if the directory already exists (at least on windows)

Reply:
fails when the destination directory already exists. Option to ignore this error only in python >=3.8

Reply:
If I'm not wrong the destination directory has a random uuid, so the probability to have already that directory should be very low. And if the destination directory already exists I don't know if the best thing to do is to rewrite it.

Reply:
the directory exists during a copy because it's created before that